packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -157,14 +149,14 @@ export class ApplicationExtension<
     protected getComponentMap?(): ComponentMap
```

Comment:
Minor: how about making it slightly easier/more direct to get the component?

Previously, it's a 2-step operation:
1. `const componentMap = this.getComponentMap()`
2. `const component = componentMap[componentName]`

```suggestion
    protected getRouteComponent?(componentName: string): React.ComponentType<any>
```

Reply:
Correct me if i'm wrong, but I believe this would make the implementation more complex for the extension developer. 

Wondering if it might be better to just have the extension developer implement returning the map while behind the scenes we handle the lookup?